framework/scripts/x86simd_generate.pl

```diff
@@ -160,6 +156,33 @@
     print ")";
 }
 
+print "\n// __attribute__ target strings for GCC and Clang";
+for (my $i = 0; $i < scalar @features; ++$i) {
+    my $feature = $features[$i];
+    my $str = $feature->{name} . ',' . $feature->{depends};
+    $str =~ s/,$//;
+    printf "#define QT_FUNCTION_TARGET_STRING_%-17s \"%s\"\n",
+        $feature->{id}, $str;
+}
+for (@architecture_names) {
+    my $arch = $architectures{$_};
+    my $base = $arch->{base};
+    my $featurestr = "";
+    if ($base ne "<>") {
+        $featurestr = "QT_FUNCTION_TARGET_STRING_ARCH_" . uc($base);
+    }
+
+    my @features = @{$arch->{features}};
+    #@features = map { defined($feature_ids{$_}) ? $_ : () } @features;
+    if (scalar @features) {
+        $featurestr .= ' ",' if length $featurestr;
+        $featurestr .= '"' unless length $featurestr;
+        $featurestr .= join(',', @features);
+        $featurestr .= '"';
+    }
+    printf "#define QT_FUNCTION_TARGET_STRING_ARCH_%-12s %s\n", uc($arch->{id}), $featurestr;
```

Comment:
QT_FUNCTION ?

Reply:
This script is shared with Qt and "QT_FUNCTION_TARGET_STRING" appears elsewhere in the script. I didn't bother to change that when we imported it and since they were never used, they didn't need fixing.

Look at the `framework/cpu_features.h` file in your build and you'll see.

Reply:
Do you want me to change this? I'd rather not.